verilog/tools/kythe/kythe_facts_extractor.h

```diff
@@ -0,0 +1,157 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_TOOLS_KYTHE_KYTHE_FACTS_EXTRACTOR_H_
+#define VERIBLE_VERILOG_TOOLS_KYTHE_KYTHE_FACTS_EXTRACTOR_H_
+
+#include <string>
+#include <utility>
+
+#include "verilog/tools/kythe/indexing_facts_tree.h"
+#include "verilog/tools/kythe/indexing_facts_tree_extractor.h"
+
+namespace verilog {
+namespace kythe {
+
+namespace {
+// Type that is used to keep track of the path to the root of indexing facts
+// tree.
+using IndexingFactsTreeContext2 = std::vector<const IndexingFactNode*>;
+
+using AutoPop2 =
+    AutoPopBack<IndexingFactsTreeContext2, const IndexingFactNode*>;
+
+}  // namespace
+
+// Node vector name for kythe facts.
+struct VName {
+  explicit VName(absl::string_view p, absl::string_view s = "",
+                 absl::string_view r = "", absl::string_view l = "verilog",
+                 absl::string_view c = "https://github.com/google/verible")
+      : signature(absl::Base64Escape(s)),
+        path(p),
+        language(l),
+        corpus(c),
+        root(r) {}
+
+  std::string ToString() const {
+    return absl::Substitute(
+        R"({"signature": "$0","path": "$1","language": "$2","root": "$3","corpus": "$4"})",
+        signature, path, language, root, corpus);
```

Comment:
move definition to .cc file to hide use of indirect libraries

would recommend defining `operator<<` for this anyhow.